scripts/lib/CIME/case.py

```diff
@@ -522,6 +513,33 @@ def _find_primary_component(self):
         return primary_component
 
 
+    def _set_info_from_primary_component(self, files, pesfile=None):
+        """
+        Sets file and directory paths that depend on the primary component of
+        this compset.
+
+        Assumes that self._primary_component has already been set.
+        """
+
+        component = self._primary_component
+
+        if pesfile is None:
+            self._pesfile = files.get_value("PES_SPEC_FILE", {"component":component})
+            pesfile_unresolved = files.get_value("PES_SPEC_FILE", {"component":component}, resolved=False)
+            logger.info("Pes     specification file is {}".format(self._pesfile))
+        else:
+            self._pesfile = pesfile
+            pesfile_unresolved = pesfile
+        self.set_lookup_value("PES_SPEC_FILE", pesfile_unresolved)
```

Comment:
There are some subtle behavior changes here compared with the previous code:

1. Previously, this wasn't being done for a compset with an alias when a pesfile was specified explicitly (causing issue #1664 )

2. Previously, for a compset without an alias without an explicitly specified pesfile, this was set with resolved = True (implicitly); now it's set with resolved = False for consistency with what happens when there is an alias. I don't think this has any important impact, but I wanted to mention it in case it's important.